js/views/modals/Settings/Page.js

```diff
@@ -37,27 +92,46 @@ export default class extends baseVw {
     this.profile.set(formData);
 
     const save = this.profile.save();
+    let saveAvatar;
+    let saveHeader;
 
     this.trigger('saving');
 
     if (!save) {
       // client side validation failed
       this.trigger('saveComplete', true);
+      this.render();
     } else {
       this.trigger('savingToServer');
-
-      save.done(() => this.trigger('saveComplete'))
-        .fail((...args) =>
-          this.trigger('saveComplete', false, true,
-            args[0] && args[0].responseJSON && args[0].responseJSON.reason || ''));
     }
 
-    // render so errrors are shown / cleared
-    this.render();
+    if (this.avatarChanged && this.avatarCropper.cropit('imageSrc')) {
```

Comment:
Can you please wrap the `if (this.avatarChanged...`, `if (this.headerChanged..` & `$.whenAll(save..` blocks in an `if (save) {` so that they're only conditionally executed if client side validation passes?

If client side validation fails, we shouldn't be kicking off any image uploads until the user fixes the errors.